src/lib.rs

```diff
@@ -1304,35 +1291,24 @@ where
     }
 
     #[inline]
-    fn write_mailbox<TX, R>(
-        &mut self,
-        idx: Mailbox,
-        tx_header: TxFrameHeader,
-        transmit: TX,
-    ) -> R
-    where
-        TX: FnOnce(&mut [u32]) -> R,
-    {
+    fn write_mailbox(&mut self, idx: Mailbox, tx_header: TxFrameHeader, buffer: &[u8]) {
         let tx_ram = self.tx_msg_ram_mut();
 
+        let tx_element = &mut tx_ram.tbsa[idx as usize];
+
         // Clear mail slot; mainly for debugging purposes.
-        tx_ram.tbsa[idx as usize].reset();
+        tx_element.reset();
+        tx_element.header.merge(tx_header);
 
-        // Calculate length of data in words
         let data_len = ((tx_header.len as usize) + 3) / 4;
 
-        //set header section
-        tx_ram.tbsa[idx as usize].header.merge(tx_header);
-
-        //set data
-        let result = transmit(&mut tx_ram.tbsa[idx as usize].data[0..data_len]);
+        let data: &[u32] = unsafe { mem::transmute(buffer) };
```

Comment:
I thought about the `from_XX_bytes`, but as it turns out that function does a transmute as well. Its only safe because of the fixed size array parameter. I'm more inclined to copy once over, its just less efficient.